apm-lambda-extension/extension/route_handlers.go

```diff
@@ -46,30 +47,33 @@ func handleInfoRequest(apmServerUrl string) func(w http.ResponseWriter, r *http.
 			return
 		}
 
-		resp, err := client.Do(req)
+		// Send request to apm server
+		serverResp, err := client.Do(req)
 		if err != nil {
 			log.Printf("error forwarding info request (`/`) to APM Server: %v", err)
 			return
 		}
-		defer resp.Body.Close()
-		body, err := ioutil.ReadAll(resp.Body)
+
+		// If WriteHeader is not called explicitly, the first call to Write
+		// will trigger an implicit WriteHeader(http.StatusOK).
+		if serverResp.StatusCode != 200 {
+			w.WriteHeader(serverResp.StatusCode)
+		}
+
+		// copy body to request sent back to the agent
+		_, err = io.Copy(w, serverResp.Body)
 		if err != nil {
 			log.Printf("could not read info request response to APM Server: %v", err)
 			return
 		}
 
-		// send status code
-		w.WriteHeader(resp.StatusCode)
-
 		// send every header received
-		for name, values := range resp.Header {
+		for name, values := range serverResp.Header {
```

Comment:
Does this work? I had assumed you need to write the headers to the response before writing the body

Reply:
Thanks for pointing that out. I've added a check in the test that the header received from the server is forwarded to the agent.